alica_engine/include/engine/BasicPlan.h

```diff
@@ -0,0 +1,22 @@
+#pragma once
+
+#include <string>
+
+namespace alica
+{
+class BasicPlan {
+public:
+    BasicPlan(const std::string& name);
+    virtual ~BasicPlan(){};
+
+    virtual void init() {};
+    virtual void onTermination() {};
+
+    const std::string& getName() const { return _name; }
```

Comment:
There is a 1:1 relationship between a Plan and a BasicPlan. Thus, BasicPlans do not need a name (Plans already have one).